src/main/java/com/microsoft/sqlserver/jdbc/dns/DNSRecordSRV.java

```diff
@@ -0,0 +1,147 @@
+            }
+            return new DNSRecordSRV(priority, weight, port, serverName);
+        }
+        catch (IllegalArgumentException err) {
+            throw err;
+        }
+        catch (Exception err) {
+            throw new IllegalArgumentException("Failed to parse DNS SRV record '" + record + "'", err);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return String.format("DNS.SRV[pri=%d w=%d port=%d h='%s']", priority, weight, port, serverName);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param priority
+     *            is lowest
+     * @param weight
+     *            1 at minimum
+     * @param port
+     *            the port of service
+     * @param serverName
+     *            the host
+     * @throws IllegalArgumentException
+     *             if priority < 0 or weight <= 1
+     */
+    public DNSRecordSRV(int priority,
+            int weight,
+            int port,
+            String serverName) throws IllegalArgumentException {
+        if (priority < 0) {
+            throw new IllegalArgumentException("priority must be >= 0, but was: " + priority);
+        }
+        this.priority = priority;
+        if (weight < 0) {
+            // Weight == 0 is OK to disable load balancing, but not below
+            throw new IllegalArgumentException("weight must be >= 0, but was: " + weight);
+        }
+        this.weight = weight;
+        if (port < 0 || port > 65535) {
+            throw new IllegalArgumentException("port must be between 0 and 65535, but was: " + port);
+        }
+        this.port = port;
+        if (serverName == null || serverName.trim().isEmpty()) {
+            throw new IllegalArgumentException("hostname is not supposed to be null or empty in a SRV Record");
+        }
+        this.serverName = serverName;
+    }
+
+    private final int weight;
+    private final int port;
+    private final String serverName;
+
+    @Override
+    public int hashCode() {
+        return serverName.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+        if (!(other instanceof DNSRecordSRV)) {
+            return false;
+        }
+
+        DNSRecordSRV r = (DNSRecordSRV) other;
+        return port == r.port && weight == r.weight && priority == r.priority && serverName.equals(r.serverName);
+    }
+
+    @Override
+    public int compareTo(DNSRecordSRV o) {
+        if (o == null) {
+            return 1;
+        }
+        int p = Integer.compare(priority, o.priority);
+        if (p != 0) {
+            return p;
+        }
+        p = Integer.compare(weight, o.weight);
+        if (p != 0) {
+            return p;
+        }
+        p = Integer.compare(port, o.port);
+        if (p != 0) {
+            return p;
+        }
+        return serverName.compareTo(o.serverName);
+    }
+
+    public int getPriority() {
```

Comment:
just wondering if user can benefit from this public methods. If so, can you please add the javaDocs for this and the rest?